R/develop.R

```diff
@@ -0,0 +1,166 @@
+
+  info$workdir <- loc$path
+  withr::with_dir(info$workdir, {
+    info <- recipe_copy_global(info, loc$config)
+    info <- recipe_copy_depends(info)
+    orderly_prepare_data(loc$config, info, parameters, envir, instance)
+  })
+
+  invisible(loc$path)
+}
+
+
+##' @export
+##' @rdname orderly_develop_start
+orderly_develop_status <- function(name = NULL, root = NULL, locate = TRUE) {
+  loc <- orderly_develop_location(name, root, locate)
+  orderly_status(loc$path)
+}
+
+
+##' @export
+##' @rdname orderly_develop_start
+orderly_develop_clean <- function(name = NULL, root = NULL, locate = TRUE) {
+  loc <- orderly_develop_location(name, root, locate)
+  status <- orderly_status(loc$path)
+  drop <- status$filename[status$derived & status$present]
+  if (length(drop) > 0L) {
+    orderly_log("remove", drop)
+    file.remove(file.path(loc$path, drop))
+  }
+  invisible(NULL)
+}
+
+
+orderly_develop_location <- function(name, root, locate) {
+  config <- orderly_config_get(root, locate)
+  config <- check_orderly_archive_version(config)
+
+  if (is.null(name)) {
+    if (!file.exists("orderly.yml")) {
+      stop("Did not find orderly.yml within working directory")
+    }
+    if (!fs::path_has_parent(getwd(), config$root)) {
+      stop("Working directory is not within the orderly root")
+    }
+    rel <- fs::path_split(fs::path_rel(getwd(), config$root))[[1]]
+    if (length(rel) != 2 || rel[[1L]] != "src") {
+      stop("Unexpected working directory - expected src/<name>")
+    }
+    name <- rel[[2L]]
+  } else if (grepl("^src/.+", name)) {
+    name <- sub("^src/", "", name)
+  }
+
+  path <- file.path(path_src(config$root), name)
+  inplace <- same_path(path, getwd())
+
+  list(config = config, name = name, path = path, inplace = inplace)
+}
+
+
+## This might be more general, but for now assume not
+orderly_status <- function(path) {
+  assert_is_directory(path, FALSE)
+  assert_file_exists(file.path(path, "orderly.yml"))
+  ## TODO: this needs making more robust
+  config <- orderly_config_get(file.path(path, "..", ".."), FALSE)
+  info <- recipe_read(path, config, FALSE)
+
+  internal <- list(orderly = "orderly.yml",
+                   script = info$script,
+                   source = info$sources,
+                   changelog = info$changelog,
+                   resource = info$resources,
+                   global = names(info$global_resources),
+                   dependency = info$depends$as,
+                   artefact = unlist(info$artefacts[, "filenames"],
+                                     FALSE, FALSE))
+  status <- data_frame(
+    filename = unlist(internal, FALSE, FALSE),
+    type = rep(names(internal), lengths(internal)))
+  extra <- setdiff(dir(path, recursive = TRUE), status$filename)
+  if (length(extra) > 0L) {
+    status <- rbind(status, data_frame(filename = extra, type = "unknown"))
+  }
+  status$present <- file.exists(file.path(path, status$filename))
+  ## TODO: better name?
```

Comment:
Old TODO still around? `derived` makes sense to me as a name

Reply:
ok, I wasn't sold but haven't come up with a better one - I'll remove the notice